### PR TITLE
Add a JSON converter for 0.7 -> 0.8 and 0.8 -> 0.9

### PR DESCRIPTION
**Pull request by electriclilies**
In this PR, I add some JSON converters. The motivation for this change is #9826-- I need to visit the virtual_device_ field in the AttrVisitor of many Relay nodes. Unfortunately, this breaks importing models serialized with TVM v 0.8. This change allows us to import those models. 

I also changed the behavior of the upgrade_json function to update 0.6 to current (0.9), 0.7 to current (0.9) and 0.8 to current (0.9). 

@mbs-octoml @tqchen @mbrookhart @mikepapadim  PTAL if you are interested! Thanks

### CODE REVIEW

File: python/tvm/ir/json_compact.py [hunk 1]
```diff
@@ -57,6 +57,53 @@ def _updater(data):
...
+    Create an update to upgrade json from v0.8 to v0.9
+
+    Returns
+    -------
+    fupdater : function
+        The updater function
+    """
+
+    def _initialize_virtual_device(item, _):
+        if "virtual_device_" not in item["attrs"].keys():
```
Comment (tqchen) [inline python/tvm/ir/json_compact.py:71]: `"virtual_device_" not in item["attrs"]` is better so it do ont have to instantiate keys.

**Review by tqchen — COMMENTED** [review]
Thanks @electriclilies , one minor nit and it LGTM

File: python/tvm/ir/json_compact.py (hunk 1 above)
Comment (electriclilies) [reply python/tvm/ir/json_compact.py]: Done! Thanks for the review

**Review by tqchen — APPROVED** [review]

**Comment by electriclilies** [comment]
Thanks @masahi @tqchen!

**Pull request merged**